src/constants.js

```diff
@@ -1,7 +1,12 @@
 import path from 'path';
-import { remote } from 'electron'; // eslint-disable-line import/no-extraneous-dependencies
+import { ipcRenderer } from 'electron';
 
-const USER_DATA = remote.app.getPath('userData');
+let USER_DATA = '';
+
+ipcRenderer.on('variable-reply', (event, arg) => {
+  USER_DATA = arg.userDataPath;
+})
+ipcRenderer.send('variable-request', 'ping');
```

Comment:
Great question. I looked into this slightly. I think this use case is here because `invoke` wasn't working and I never went back and switched it. I'll look into it more but my take away is:

`on / send` is always live and listening and will take action on any `variable-reply`.

`invoke` is more of a component requesting something once because this component or function was called to do something.

Probably in most our cases `invoke` makes sense. I can go back and look to see if there's any reason to keep some of the `on / send` calls.

Reply:
After a little more looking into this I think that `invoke / handler` is new syntax to better help with `promises` and using `async / await`. Though `send / on` could be useful if you had something updating another state like a countdown or progress bar or something.